From b91b76e959a699d7897a7e99323f6a780297ba33 Mon Sep 17 00:00:00 2001
From: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Date: Fri, 2 Apr 2021 20:31:46 +0300
Subject: [PATCH] armtf: Clarify check for the "h3ulcb" based machines in
 do_extra_ipl_opt()

Current check in do_extra_ipl_opt() doesn't recognize "h3ulcb" based machines
which MACHINE_ARCH is not exact "h3ulcb" (h3ulcb-4x2g, h3ulcb-4x2g-kf).
This leads to the fact that extra build for bl2 is performed without
RCAR_GEN3_ULCB=1 option. So, clarify check to cover all machines which
MACHINE_ARCH contains "h3ulcb" string.

Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
---
 .../recipes-bsp/arm-trusted-firmware/arm-trusted-firmware_git.bb        | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meta-rcar-gen3/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware_git.bb b/meta-rcar-gen3/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware_git.bb
index d26b5d1..56039ef 100644
--- a/meta-rcar-gen3/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware_git.bb
+++ b/meta-rcar-gen3/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware_git.bb
@@ -103,7 +103,7 @@ python do_extra_ipl_opt () {
     machine = d.getVar('MACHINE_ARCH')
 
     if soc == "r8a7795":
-        if machine != "h3ulcb":
+        if "h3ulcb" not in machine:
             do_extra_aft_build(d, "H3")
         else:
             do_extra_aft_build(d, "H3ULCB")
-- 
2.7.4

